internal/services/contracts_manufacturer_service.go

```diff
@@ -0,0 +1,61 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/DIMO-Network/identity-api/internal/config"
+	"github.com/DIMO-Network/identity-api/internal/contracts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+type ManufacturerContractService struct {
+	log              *zerolog.Logger
+	settings         *config.Settings
+	registryInstance *contracts.Registry
+}
+
+func NewManufacturerContractService(log *zerolog.Logger,
+	settings *config.Settings,
+	client *ethclient.Client) (*ManufacturerContractService, error) {
+
+	contractAddress := common.HexToAddress(settings.DIMORegistryAddr)
+	registryInstance, err := contracts.NewRegistry(contractAddress, client)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed instance NewRegistry: %s", err)
+	}
+
+	return &ManufacturerContractService{
+		log:              log,
+		settings:         settings,
+		registryInstance: registryInstance,
+	}, nil
+}
+
+func (m *ManufacturerContractService) GetTableName(ctx context.Context, manufacturerName string) (*string, error) {
```

Comment:
In general, I think we try to not call the contract in this repo. Is there some event or something we can listen to instead of doing this?

Reply:
What I would do is listen for these events:

https://github.com/DIMO-Network/dimo-identity/blob/e16ed5a88ff61a975e7927db83f5f465eb3b1365/contracts/implementations/tableland/DeviceDefinitionTable.sol#L28-L36

We could then maintain a column in the manufacturers table that knows the NFT id for the Tableland table corresponding to a manufacturer. Then the table name is just `_{chainId}_{tokenId}`.

Reply:
@LorranSutter It would be nice if creating the table also emitted `ManufacturerTableSet`, don't you think? Then we'd only have to listen for one event.